pyro/infer/mcmc/logger.py

```diff
@@ -0,0 +1,117 @@
+import json
+import logging
+import os
+import sys
+from collections import OrderedDict
+
+from tqdm.autonotebook import tqdm
+
+LOG_MSG = "LOG"
+TQDM_MSG = "TQDM"
+
+
+class TqdmHandler(logging.StreamHandler):
+    """
+    Handler that synchronizes the log output with the
+    :class:`~tqdm.tqdm` progress bar.
+    """
+    def emit(self, record):
+        try:
+            msg = self.format(record)
+            self.flush()
+            tqdm.write(msg, file=sys.stderr)
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except Exception:
+            self.handleError(record)
+
+
+class MCMCLoggingHandler(logging.Handler):
```

Comment:
This seems more general than `MCMC`, e.g. I often use `sys.stdout.write('.');sys.stdout.flush()` to simulate progress bar for SVI in notebooks. Would it be reasonable to move this file to `pyro.infer.logger` and rename this class `InferenceLoggingHandler` or something, so we can use it for other inference algorithms? Or should I simply use `tqdm` directly in those notebooks?

Reply:
Ah, I see that although this class is general, its usage is specialized to MCMC below. Fine to leave it here.

Reply:
That's a good idea. Let me try to refactor this to make it more general.